src/app/util/ember-compatibility-functions.cpp

```diff
@@ -521,16 +518,10 @@ CHIP_ERROR ReadSingleClusterData(FabricIndex aAccessingFabricIndex, const Concre
     Protocols::InteractionModel::Status imStatus = ToInteractionModelStatus(emberStatus);
     if (imStatus == Protocols::InteractionModel::Status::Success)
     {
-        // TODO: Add DataVersion support
-        attributeDataIBBuilder.DataVersion(kTemporaryDataVersion).EndOfAttributeDataIB();
-        ReturnErrorOnFailure(attributeDataIBBuilder.GetError());
-    }
-    else
-    {
-        ReturnErrorOnFailure(SendFailureStatus(aPath, aAttributeReport, imStatus, backup));
+        SendSuccessStatus(attributeDataIBBuilder);
```

Comment:
```suggestion
        return SendSuccessStatus(attributeDataIBBuilder);
```
Otherwise we'll fall through to the SendFailureStatus call.

Reply:
Good catch, yes all SendXXXStatus should be return.